beacon-chain/db/filesystem/save_blob.go

```diff
@@ -0,0 +1,88 @@
+// The filesystem package provides a mechanism for managing and storing binary blob data.
+// It offers functionalities for saving and ensuring the integrity of blobs using SHA-256 checksums.
+// In order to maintain data integrity we first write to a temporary file and then atomically
+// rename the file to its final destination.
+
+package filesystem
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"os"
+	"path"
+
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/prysm/v4/io/file"
+	ethpb "github.com/prysmaticlabs/prysm/v4/proto/prysm/v1alpha1"
+)
+
+type BlobStorage struct {
+	baseDir string
+}
+
+// SaveBlobData saves blobs given a list of sidecars.
+func (bs *BlobStorage) SaveBlobData(sidecars []*ethpb.BlobSidecar) error {
+	if len(sidecars) == 0 {
+		return errors.New("no blob data to save")
+	}
+	for _, sidecar := range sidecars {
+		blobPath := path.Join(bs.baseDir, fmt.Sprintf(
+			"%d_%x_%d_%x.blob",
+			sidecar.Slot,
+			sidecar.BlockRoot,
+			sidecar.Index,
+			sidecar.KzgCommitment,
+		))
```

Comment:
This will be a good helper, where input the sidecar and get a path. Less chance to someone mess up